apps/olat-api/src/index.ts

```diff
@@ -0,0 +1,488 @@
+  }))
+
+  return coursesDetails
+}
+
+app.get('/api/configuration/courses', (req: Request, res: Response) => {
+  const headerStatusCode = checkHeader(req)
+  if (headerStatusCode !== StatusCode.SUCCESS) {
+    return res
+      .status(headerStatusCode)
+      .json({ error: 'Invalid request headers' })
+  }
+
+  // verify that a valid providerAccountId is provided
+  const providerAccountId = req.query.identityMappingIdentifier
+  if (!providerAccountId || typeof providerAccountId !== 'string') {
+    return res
+      .status(StatusCode.BAD_REQUEST)
+      .json({ error: 'Missing providerAccountId' })
+  }
+
+  // extract the provider from the providerAccountId
+  const provider = providerAccountId.split('@')[1]?.split('.')[0]
+  if (!provider || typeof provider !== 'string') {
+    return res
+      .status(StatusCode.BAD_REQUEST)
+      .json({ error: 'Extraction of provider from providerAccountId failed' })
+  }
+
+  getCourses(provider, providerAccountId)
+    .then((courses) => {
+      if (courses?.length === 0) {
+        return res
+          .status(StatusCode.NOT_FOUND)
+          .json({ error: 'No courses found for this user' })
+      }
+
+      res.set('Content-Type', 'application/json')
+      return res.status(StatusCode.SUCCESS).json({
+        courses: courses,
+        timestamp: new Date().toISOString(),
+        api: API_NAME,
+      })
+    })
+    .catch((error) => {
+      console.error('Error fetching courses:', error)
+      return res
+        .status(StatusCode.INTERNAL_SERVER_ERROR)
+        .json({ error: 'Internal server error' })
+    })
+})
+
+async function getActivityTypes(): Promise<any[] | null> {
+  // filter out fields
+  return activityTypesAvailable.map((activityType) => ({
+    ...pick(activityType, [
+      'id',
+      'path',
+      'olatConfigurationKey',
+      'isEmailTransferRequired',
+    ]),
+  }))
+}
+
+app.get('/api/configuration/activityTypes', (req: Request, res: Response) => {
+  const headerStatusCode = checkHeader(req)
+  if (headerStatusCode !== StatusCode.SUCCESS) {
+    return res
+      .status(headerStatusCode)
+      .json({ error: 'Invalid request headers' })
+  }
+
+  getActivityTypes()
+    .then((activityTypes) => {
+      if (activityTypes === null) {
+        return res
+          .status(StatusCode.NOT_FOUND)
+          .json({ error: 'Course not found' })
+      }
+
+      res.set('Content-Type', 'application/json')
+      return res.status(StatusCode.SUCCESS).json({
+        activityTypes: activityTypes,
+        timestamp: new Date().toISOString(),
+        api: API_NAME,
+      })
+    })
+    .catch((error) => {
+      console.error('Error fetching activity types:', error)
+      return res
+        .status(StatusCode.INTERNAL_SERVER_ERROR)
+        .json({ error: 'Internal server error' })
+    })
+})
+
+async function getCourseActivityTypes(courseId: string): Promise<any[] | null> {
+  const course = await prisma.course.findUnique({
+    where: { id: courseId },
```

Comment:
we also need to make sure that this user can access the course with the specified ID. so we need to ensure that there are derived permissions for the user to access the course. this means also passing the provider account id here via a POST request.